src/app/code/community/Omise/Gateway/Model/Payment.php

```diff
@@ -68,6 +88,15 @@ public function getAmountInSubunits($amount, $currency)
      */
     protected function _process(Varien_Object $payment, $params)
     {
+        // check if we need to create a source
+        if (!empty($params['source']) && is_array($params['source'])) {
+            $source = $this->_makeSource(array_merge($params['source'], [
```

Comment:
@jonrandy actually I just remembered that we don't need to make a Source object before creating a new charge but just simply past `'source' => [ 'type' => 'alipay' ]` so we don't need to create 2 requests (for creating Source and Charge objects)

Reply:
That's not what the docs say - https://www.omise.co/alipay